gap/oper.gi

```diff
@@ -1771,3 +1771,106 @@ function(digraph, k)
 
   return DigraphByAdjacencyMatrix(out);
 end);
+
+#
+
+InstallMethod(IsMatching,
+"for a digraph, and a list",
+[IsDigraph, IsList],
+function(gr, edges)
+  # Note this will return false if you have entered the same edge twice,
+  # and considers edges as undirected.  Thus [e_1,e_2] = [e_2,e_1].
+  # Similarly, the function considers the symmetric closure of the input
+  # graph. Loops are not considered adjacent to themselves.
+  # Thus a matching may exist with loops.
+
+  local e, f, prob, gredges;
+
+  #gredges will contain the edges of gr
+  gredges := DigraphEdges(gr);
+
+  # Prob will hold the edges in our edge set we haven't yet checked
+  # (ie the potential problem edges)
+  prob := ShallowCopy(edges);
+
+  for e in edges do
```

Comment:
So essentially, this method checks whether each 'edge' in `edges` is an edge of the digraph (or the reverse of an edge), and then checks that no vertex appears in more than one edge?